### PR TITLE
Remove override of navigationbar style for LoginViewController

### LINKED ISSUE #3018: [Xcode 12] Fix preferredStatusBarStyle override warning
This warning is shown when compiling with Xcode 12. 

```
ld: warning: method '-preferredStatusBarStyle' in category from /Users/shiki/Library/Developer/Xcode/DerivedData/WooCommerce-gbnqhhlybzucibaxaxxixpgqmqok/Build/Intermediates.noindex/WooCommerce.build/Debug-iphonesimulator/WooCommerce.build/Objects-normal/x86_64/LoginNavigationController+Woo.o overrides method from class in /Users/shiki/Library/Developer/Xcode/DerivedData/WooCommerce-gbnqhhlybzucibaxaxxixpgqmqok/Build/Products/Debug-iphonesimulator/WordPressAuthenticator/WordPressAuthenticator.framework/WordPressAuthenticator(LoginNavigationController.o)
```

### PR DESCRIPTION
**Pull request by ctarda**
Closes #3018 

The warning is issued because LoginViewController+Woo overrides `preferredStatusBarStyle`.

The override is accompanied by this comment:

```
    /// TODO: Create a property in Authenticator then delete this
    /// https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/26
```

And it turns out the referenced issue wordpress-mobile/WordPressAuthenticator-iOS#26 has already been closed. So it should be possible to remove the override on our side.

| Before | After |
| ---- | ---- |
|  |  |

# What's new
* Remove LoginViewcontroller+Woo
* Update the status bar style provided to the authenticator style to `lightContent`, as WPAuthenticator would take and apply this style.

# How to test
* Checkout the branch, build and run
* Log out if necessary, navigate to the login flow.
* Check the status bar style is still light.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/AuthenticationManager.swift [hunk 1]
```diff
@@ -72,7 +72,7 @@ class AuthenticationManager: Authentication {
                                                 navBarBadgeColor: .primary,
                                                 navBarBackgroundColor: .appBar,
                                                 prologueTopContainerChildViewController: LoginPrologueViewController(),
-                                                statusBarStyle: .default)
+                                                statusBarStyle: .lightContent)
```
Comment (jaclync) [inline WooCommerce/Classes/Authentication/AuthenticationManager.swift]: I think this line of change might have caused the status bar on the prologue screen to be invisible in Light mode:

(the time and network/battery status are not visible)

PR branch | `develop` or if I changed this line back to `.default`
-- | --
Simulator Screen Shot - iPhone 11 - 2020-11-05 at 09 59 49 | Simulator Screen Shot - iPhone 11 - 2020-11-05 at 10 00 20

File: WooCommerce/Classes/Extensions/LoginNavigationController+Woo.swift [hunk 2]
```diff
@@ -1,12 +0,0 @@
-import UIKit
-import WordPressAuthenticator
-
-extension LoginNavigationController {
-
-    /// TODO: Create a property in Authenticator then delete this
-    /// https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/26
```
Comment (jaclync) [inline WooCommerce/Classes/Extensions/LoginNavigationController+Woo.swift:7]: nice getting rid of this override 👍

**Review by jaclync — APPROVED** [review]
Just a comment on the status bar on the prologue screen, otherwise LGTM!

I tested the login flow in both Light and Dark mode, and verified that the warning is gone ✅

**Comment by ctarda** [comment]
Good catch @jaclync , thanks!

I pushed a commit to revert that change to the style, now it looks exactly as develop:

| Prologue | Login |
| ---- | ---- |
|  |  | 

I'll merge as soon as CI gives me the ✅

**Pull request merged**